Engine/DataFeeds/FileSystemDataFeed.cs

```diff
@@ -360,12 +367,37 @@ private Subscription CreateUniverseSubscription(SubscriptionRequest request)
         }
 
         /// <summary>
-        /// Creates the correct enumerator factory for the given request
+        /// Creates the correct enumerator factory for the given request.
+        /// Returns null if a subscription should not be created for this request.
         /// </summary>
         private ISubscriptionEnumeratorFactory GetEnumeratorFactory(SubscriptionRequest request)
         {
             if (request.IsUniverseSubscription)
             {
+                var benchmarkUniverse = request.Universe as BenchmarkUniverse;
+                if (benchmarkUniverse != null)
+                {
+                    // Trigger universe selection when benchmark security added
+                    benchmarkUniverse.CollectionChanged += (sender, args) =>
+                    {
+                        if (args.Action != NotifyCollectionChangedAction.Add) return;
+
+                        var items = args.NewItems;
+                        if (items == null) return;
+
+                        var symbol = items.OfType<Symbol>().FirstOrDefault();
+                        if (symbol == null) return;
+
+                        var collection = new BaseDataCollection(_algorithm.UtcTime, symbol);
```

Comment:
I believe that if `Symbol` is of security type `future` or `option`, `collection` should be of type `FuturesChainUniverseDataCollection` or `OptionChainUniverseDataCollection`, else will throw.
Check this guy https://github.com/QuantConnect/Lean/blob/master/Engine/AlgorithmManager.cs#L811